src/TypeInference/Pattern.ml

```diff
@@ -140,7 +140,68 @@ let get_ctor_info ~pos ~env (cpath : S.ctor_name S.path S.node) tp =
   | Whnf_Effect _ | Whnf_Effrow _ ->
     failwith "Internal kind error"
 
-let rec check_ctor_named_args ~pos ~env ~scope nps named =
+(** Introduce all named types and named arguments into the environment.
+  The type variables are refreshed, so the function returns a renaming. *)
+let open_named ~pos ~public env targs named =
```

Comment:
This function is long, and could be improved by defining two helper functions: one for types, and one for values.